src/expr/interpret/eval.hpp

```diff
@@ -120,6 +120,10 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       case Error::SCHEMA_VERTEX_SECONDARY_LABEL_IS_PRIMARY:
       case Error::SCHEMA_VERTEX_PRIMARY_PROPERTIES_UNDEFINED:
         throw ExpressionRuntimeException("Unexpected schema violation when accessing {}.", accessed_object);
+      case Error::MISMATCHED_SHARD_VERSION:
+        throw ExpressionRuntimeException(
+            "Cluster performed a Shard split or merge that invalidated the transaction's metadata. This should have "
+            "been handled in the RequestRouter.");
```

Comment:
This part of the code is only used on the shards, therefore handling this error in the request router doesn't make sense.